### PR TITLE
[opt](Nereids) translate "between" to "equalTo" if upper equals to lower

### PR DESCRIPTION
**Pull request by englefly**
### What problem does this PR solve?
predicate "A between 1 and 10" is converted to "A<=10 and A>=1"
but if the lower value equals to upper value, between predicate can be converted to a euqalTo predicate. 
Example:
"A between 1 and 1"  is converted to "A=1"

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Check List (For Committer)

- Test 

    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [x] No.
    - [ ] Yes. 

- Does this need documentation?

    - [ ] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/test/java/org/apache/doris/nereids/parser/BetweenTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class BetweenTest {
+    private static final NereidsParser PARSER = new NereidsParser();
+
+    @Test
+    public void testBetween() {
+        String expression ="A between 1 and 1"; //
```
Comment (morrySnow) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/parser/BetweenTest.java:32]: should test more expression, such as unboundSlot, Case when, etc..

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java [hunk 2]
```diff
@@ -3495,10 +3495,16 @@ private Expression withPredicate(Expression valueExpression, PredicateContext ct
             Expression outExpression;
             switch (ctx.kind.getType()) {
                 case DorisParser.BETWEEN:
-                    outExpression = new And(
-                            new GreaterThanEqual(valueExpression, getExpression(ctx.lower)),
-                            new LessThanEqual(valueExpression, getExpression(ctx.upper))
-                    );
+                    Expression lower = getExpression(ctx.lower);
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:3498]: support the following case: dt (date type)BETWEEN '2024-08-19' AND       '2024-08-19 14:44:28' =>  BETWEEN '2024-08-19' AND  '2024-08-19' => dt = '2024-08-19'

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**